src/command.cpp

```diff
@@ -47,6 +47,53 @@ const char* Command::getstring() const
     return json.c_str();
 }
 
+//return true when the response is an error, false otherwise (in that case it doesn't consume JSON chars)
+bool Command::checkError(Error& errorDetails, JSON& json)
+{
+    error e;
+    if (json.isNumericError(e))
+    {
+        errorDetails.setErrorCode(e);
+        return true;
+    }
```

Comment:
I see isNumericError was added rather than using the already isnumeric() and getint().   The difference seems to be whether numbers above 0 would be included.   The old code for commands I've checked would just store the positive number in the Error and continue with that, I think we could do the same here without adding a new function in the JSON class that has overlapping functionality with existing functions - unless there are commands where we do return just a single positive number on success? (but that would be an error prone design)  thanks

Reply:
Yes, I think so, there are several commands that return a single positive number on success:
CommandMediaCodecs, CommandCreditCardQuerySubscriptions, CommandGetPaymentMethods (this one, can return several positive number)
In `checkError` we have tried to handle only negative plus 0 numbers. We thought that it is not very correct semantically, return a positive number in a function `checkError`

Reply:
I see, thanks for the info - of course if those commands already do that then we have to support it.   By the way, reading CommandGetPaymentMethods, I see that 0 returned is actually data to be processed rather than an API_OK value which makes the code quite awkward but it looks correct.  I wonder if an option for checkError() to not consume 0 would help with those cases.  Anyway, I understand the reasoning, thanks for that.  cheers